ngraph/cmake/patches/onnx_patch.diff

```diff
@@ -0,0 +1,30 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 5254f7ee..e7a0ce2b 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -380,7 +380,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
+   # So, create a object library
+   add_library(onnx OBJECT ${ONNX_SRCS})
+ else()
+-  add_library(onnx ${ONNX_SRCS})
++  # onnx target doesn't export symbols
++  add_library(onnx STATIC ${ONNX_SRCS})
+ endif()
+ 
+ target_include_directories(onnx PUBLIC
+diff --git a/onnx/onnx_pb.h b/onnx/onnx_pb.h
+index 7dc68dea..c932b806 100644
+--- a/onnx/onnx_pb.h
++++ b/onnx/onnx_pb.h
+@@ -40,9 +40,10 @@
+ //
+ // This solution is similar to
+ // https://github.com/pytorch/pytorch/blob/master/caffe2/core/common.h
+-#if defined(ONNX_BUILD_SHARED_LIBS) || defined(ONNX_BUILD_MAIN_LIB)
```

Comment:
why do we need to remove `ONNX_BUILD_SHARED_LIBS`? We build `onnx` as a static

Reply:
`onnx_pb.h` header is responsible only for setting `onnx_proto` (files generated by protobuf) linking attributes (by `ONNX_API` decorator).

`onnx` target components (some features to check models, perform onnx-level shape inference etc.) doesn't use `ONNX_API` macro - or to be more clear - they don't control linking (such as dll export) in any way.
It was one of the reason why we don't want to have `onnx` target shared.
We can use to this purpose [WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/v3.4/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html) but it is probably not very clear way. What's more shared `onnx` target is not needed.